### PR TITLE
Avoid registering babel plugin multiple times.

### LINKED ISSUE #262: v4.1.2 of ember-template-imports may register babel plugin multiple times
In https://github.com/ember-cli/ember-template-imports/pull/241/files#diff-70031c11e5dceb900db185b04dd050b0f3090e2dc6bee7b0948726419ce91b0eR19 we added registration for babel plugin.

In case of our app, we have a monorepo with bunch of v1 addons, engines and the host app which all include `ember-template-imports` in its `dependencies` and/or `devDependencies`.

When we tried to bump `ember-template-imports` to v4.1.2, we could not due to such errors:

```
error	07-Oct-2024 22:58:23	#19 105.6 Build Error (Babel) in XXX/test-support/helpers/setup-experiments.ts
error	07-Oct-2024 22:58:23	#19 105.6 
error	07-Oct-2024 22:58:23	#19 105.6 Duplicate plugin/preset detected.
error	07-Oct-2024 22:58:23	#19 105.6 If you'd like to use two separate instances of a plugin,
error	07-Oct-2024 22:58:23	#19 105.6 they need separate names, e.g.
error	07-Oct-2024 22:58:23	#19 105.6 
error	07-Oct-2024 22:58:23	#19 105.6   plugins: [
error	07-Oct-2024 22:58:24	#19 105.6     ['some-plugin', {}],
error	07-Oct-2024 22:58:24	#19 105.6     ['some-plugin', {}, 'some unique name'],
error	07-Oct-2024 22:58:24	#19 105.6   ]
error	07-Oct-2024 22:58:24	#19 105.6 
error	07-Oct-2024 22:58:24	#19 105.6 Duplicates detected are:
error	07-Oct-2024 22:58:24	#19 105.6 [
error	07-Oct-2024 22:58:24	#19 105.6   {
error	07-Oct-2024 22:58:24	#19 105.6     "alias": "/XXX/node_modules/.pnpm/ember-template-imports@4.1.2/node_modules/ember-template-imports/src/babel-plugin.js",
error	07-Oct-2024 22:58:24	#19 105.6     "options": {
error	07-Oct-2024 22:58:24	#19 105.6       "v": 1,
error	07-Oct-2024 22:58:24	#19 105.6       "root": "/XXX/packages/XYZ"
error	07-Oct-2024 22:58:24	#19 105.6     },
error	07-Oct-2024 22:58:24	#19 105.6     "dirname": "/XXX/apps/XYZ",
error	07-Oct-2024 22:58:24	#19 105.6     "ownPass": false,
error	07-Oct-2024 22:58:24	#19 105.6     "file": {
error	07-Oct-2024 22:58:24	#19 105.6       "request": "/XXX/node_modules/.pnpm/ember-template-imports@4.1.2/node_modules/ember-template-imports/src/babel-plugin.js",
error	07-Oct-2024 22:58:24	#19 105.6       "resolved": "/XXX/node_modules/.pnpm/ember-template-imports@4.1.2/node_modules/ember-template-imports/src/babel-plugin.js"
error	07-Oct-2024 22:58:24	#19 105.6     }
error	07-Oct-2024 22:58:24	#19 105.6   },
error	07-Oct-2024 22:58:24	#19 105.6   {
error	07-Oct-2024 22:58:24	#19 105.6     "alias": "/XXX/node_modules/.pnpm/ember-template-imports@4.1.2/node_modules/ember-template-imports/src/babel-plugin.js",
error	07-Oct-2024 22:58:24	#19 105.6     "options": {
error	07-Oct-2024 22:58:24	#19 105.6       "v": 1,
error	07-Oct-2024 22:58:24	#19 105.6       "root": "/XXX/packages/XYZ"
error	07-Oct-2024 22:58:24	#19 105.6     },
error	07-Oct-2024 22:58:24	#19 105.6     "dirname": "/XXX/apps/XYZ",
error	07-Oct-2024 22:58:24	#19 105.6     "ownPass": false,
error	07-Oct-2024 22:58:24	#19 105.6     "file": {
error	07-Oct-2024 22:58:24	#19 105.6       "request": "/XXX/node_modules/.pnpm/ember-template-imports@4.1.2/node_modules/ember-template-imports/src/babel-plugin.js",
error	07-Oct-2024 22:58:24	#19 105.6       "resolved": "/XXX/node_modules/.pnpm/ember-template-imports@4.1.2/node_modules/ember-template-imports/src/babel-plugin.js"
error	07-Oct-2024 22:58:24	#19 105.6     }
error	07-Oct-2024 22:58:24	#19 105.6   },
...
```

and few hundred lines of the same error.

### PR DESCRIPTION
**Pull request by SergeAstapov**
Fixes #262.

Suggested changes inspired by https://github.com/ember-cli/ember-cli-htmlbars/blob/master/lib/ember-addon-main.js#L58-L59

### CODE REVIEW

File: ember-addon-main.js [hunk 1]
```diff
@@ -1,6 +1,23 @@
 'use strict';
 
 let VersionChecker = require('ember-cli-version-checker');
+let babelPluginPath = require.resolve('./src/babel-plugin.js');
+
+function isAddTOCNamesForInspectorBabelPluginRegistered(plugins, root) {
```
Comment (NullVoxPopuli) [inline ember-addon-main.js:6]: why is this function called this? what's a TOCNames for Inspector?

**Review by NullVoxPopuli — APPROVED** [review]
Tests pass.
Confused about the name of a (long) function, but otherwise, things make sense.

File: ember-addon-main.js (hunk 1 above)
Comment (SergeAstapov) [reply ember-addon-main.js:6]: TOC is for Template Only Component, however we can name it whatever it makes sense, maybe we could do just `isBabelPluginRegistered` for simplicity as it's the only plugin (if I'm not mistaken)? WDYT @NullVoxPopuli ?

Comment (NullVoxPopuli) [reply ember-addon-main.js:6]: ya I like that gives just the information we need

**Pull request merged**